subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/tooling/BuildServices.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.tooling
+
+import groovy.lang.Closure
+import org.gradle.api.Action
+import org.gradle.internal.service.scopes.ExceptionCollector
+import java.util.Arrays
+import java.util.Collections
+import java.util.concurrent.CopyOnWriteArrayList
+
+
+internal
+object BuildServices {
+
+    private
+    val NOOP_COLLECTOR: ExceptionCollector = object : ExceptionCollector {
+        override fun stop() {}
+        override fun getExceptions(): List<Exception> = emptyList()
+        override fun addException(e: Exception) {}
+        override fun <T> decorate(action: Action<T>): Action<T> = action
+        override fun <T> decorate(closure: Closure<T>): Closure<T> = closure
+    }
+
+    @Suppress("unused")
+    fun createExceptionCollector(): ExceptionCollector {
```

Comment:
:x: this can be moved to `org.gradle.kotlin.dsl.provider.BuildServices` and there's no need for a new `.tooling` package nor for a new service provider type

Reply:
Thanks for showing me the way. I'm less comfortable with the .kt codebase in gradle/gradle.